Update to v2.3.1

maxderhak opened the pull request:
Added support for hybrid ICC/iccMAX profiles with support for spectral images in iccAppyProfiles, and spectral reproduction in iccApplySearch using new CIccCmmSearch object with other example profiles.  Also added tool to create V4 display profiles using v5 display profiles with custom observer profiles

maxderhak commented:
@xsscx I'm going to need your help getting this to work with all the changes that you have made.  There are a few more tools and new objects added to IccProfLib.

xsscx reviewed (changes requested):
# Manual Review

## Reproduction for PR170

```
git clone https://github.com/InternationalColorConsortium/iccDEV.git
cd iccDEV
git fetch origin pull/170/head:pr-170
git checkout pr-170
cd Build
cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -DCMAKE_BUILD_TYPE=Release -DENABLE_TOOLS=ON  -Wno-dev Cmake/
make clean
make -j$(nproc)
```

## Result
All Builds Fail

## Summary

There is a new Tool iccSearch to be config'd with CmakeLists.txt - dh
There are 3 files to add to ProfLib: IccSearch.h, IccCmmSearch.cpp & IccCmmSearch.h - dh

Housekeeping: 
restore from master existing workflows - dh
restore from master Build/Cmake dir - dh 
harvest any changes hardcoded in VS files - dh

xsscx commented:
@maxderhak I'll work on config & compile soon.

xsscx commented:
@maxderhak 

## Status

I cloned the master branch, copied the 3 source files to IccProfLib, added the Tools & Cmake directories for iccApplySearch then modified the CMakeLists.txt files. Only 2 errors to report so far:

```
...
[ 96%] Built target iccApplyNamedCmm
/home/xss/pr170/repro2/iccDEV/Tools/CmdLine/IccApplySearch/iccApplySearch.cpp:233:3: error: unknown type name 'CIccCfgSearchApply'
  233 |   CIccCfgSearchApply cfgSearchApply;
      |   ^
/home/xss/pr170/repro2/iccDEV/Tools/CmdLine/IccApplySearch/iccApplySearch.cpp:376:5: error: unknown type name 'CIccCfgPccWeight'
  376 |     CIccCfgPccWeight* pPccWeight = pPccIter->get();
      |     ^
```

I will push my Cmake config updates to the pr171 branch at URL https://github.com/InternationalColorConsortium/iccDEV/tree/pr171 shortly, setting up a matrix runner now....

xsscx commented:
Runner https://github.com/xsscx/PatchIccMAX/actions/workflows/pr171.yaml

xsscx commented:
# Summary

@maxderhak Code revisions for this PR on pr171 branch at URL https://github.com/InternationalColorConsortium/iccDEV/tree/pr171

Branch pr171 compiles cross platform and the unix reproduction is:

```
git clone https://github.com/InternationalColorConsortium/iccDEV.git
cd iccDEV
git checkout pr171
cd Build
cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -DCMAKE_BUILD_TYPE=Release -DENABLE_TOOLS=ON  -Wno-dev Cmake/
make clean
make -j$(nproc)
```

## NET changes
 - Removed all changes to the .github/workflows directory
 - Removed all VS/Sln/vcxproj changes
 - Added the Cmake Configs

## TODO

- Update this PR170 with the changes seen on pr171
   - OR Close this PR and use pr171 branch with fresh PR.
- Confirm the project name for the new code be modified to `iccApplySearch` ? 
  - Currently: project(iccApplyReverse)
- Suggestion to add the license / copyright block to IccSearch.h

xsscx commented:
@maxderhak Nice Work!

```
E:\pr171\iccdev\Testing\hybrid>iccFromXml.exe Data\Spec380_10_730-D50_2deg.xml ICC\Spec380_10_730-D50_2deg.icc
Profile parsed and saved correctly

************************************************
Make a multi-spectral image from a spectral one
************************************************

E:\pr171\iccdev\Testing\hybrid>iccTiffDump.exe Data\smCows380_5_780.tif
TIFFReadDirectory: Warning, Sum of Photometric type-related color channels and ExtraSamples doesn't match SamplesPerPixel. Defining non-color channels as ExtraSamples..
-------------------->Tiff Image Dump<---------------------------
Filename:          Data\smCows380_5_780.tif
Size:              (600 x 420) pixels, (8.33" x 5.83")
Planar:            Interleaved samples
BitsPerSample:     16
SamplesPerPixel:   81
Photometric:       Min Is Black
BytesPerLine:      97200
Resolution:        (72.000000 x 72.000000) pixels per/inch
Compression:       LZW
Profile:           Embedded
 Version:          5.00
 Color Space:      0x0051ChannelData
 Spectral PCS:     0x0051ChannelReflectanceData
 Spectral Range:   start=380.0nm, end=780.0nm, steps=81
 Description:      380nm to 780nm in 5nm steps spectral reflectance using D50 and standard 2deg observer with Wpt MAT based PCC

E:\pr171\iccdev\Testing\hybrid>iccApplyProfiles.exe Data\smCows380_5_780.tif Results\MS_smCows.tif 2 1 0 1 1 -embedded 3 ICC\MultSpectralRGB.icc 10003
TIFFReadDirectory: Warning, Sum of Photometric type-related color channels and ExtraSamples doesn't match SamplesPerPixel. Defining non-color channels as ExtraSamples..
100%

E:\pr171\iccdev\Testing\hybrid>iccTiffDump.exe Results\MS_smCows.tif
-------------------->Tiff Image Dump<---------------------------
Filename:          Results\MS_smCows.tif
Size:              (600 x 420) pixels, (8.33" x 5.83")
Planar:            Samples in separate planes
BitsPerSample:     16
SamplesPerPixel:   8
ExtraSamples:      5
Photometric:       RGB
BytesPerLine:      9600
Resolution:        (72.000000 x 72.000000) pixels per/inch
Compression:       LZW
Profile:           Embedded
 Version:          4.40
 Color Space:      RgbData
 Colorimetric PCS: XYZData
 Description:      RGB view of MultiSpectral 8 Channel Data
 Sub-Profile:      Embedded
  Version:          5.10
  Color Space:      MCH8Data/8ColorData
  Spectral PCS:     0x0024ChannelReflectanceData
  Spectral Range:   start=380.0nm, end=730.0nm, steps=36
  Description:      Eight Channel Abridged Reflectance Encoder/Decoder
*****************************************************************
Apply PCC's to the spectral images to get colorimetric renderings
*****************************************************************
```

maxderhak commented:
> ## TODO
> * Update this PR170 with the changes seen on pr171 - md
> * Suggestion to add the license / copyright block to IccSearch.h - md
> 
I've done the above.  I also changed the namespace from refIccMAX to iccDEV and updated a few more of the licenses in code.
I'm hoping that the merge now makes sense.  Thanks @xsscx !
